src/Bicep.Core/Parsing/ParamsParser.cs

```diff
@@ -74,13 +75,42 @@ public override ProgramSyntax Program()
         private UsingDeclarationSyntax UsingDeclaration()
         {
             var keyword = ExpectKeyword(LanguageConstants.UsingKeyword);
+
+            SyntaxBase expression = reader.Peek().Type switch
+            {
+                TokenType.EndOfFile or
+                TokenType.NewLine or
+                TokenType.NoneKeyword => WithRecovery(
+                    () => new NoneLiteralSyntax(
+                        Expect(
+                            TokenType.NoneKeyword,
+                            e => e.ExpectedKeyword(LanguageConstants.NoneKeyword)
+                        )
+                    ),
+                    GetSuppressionFlag(keyword),
+                    TokenType.NewLine
+                ),
```

Comment:
If you remove the checks for EOF & newline, this could be simplified to remove the `WithRecovery`:
```csharp
TokenType.NoneKeyword => new NoneLiteralSyntax(ExpectKeyword(TokenType.NoneKeyword)),
```

Reply:
As you suggested I simplified the code like this;

```csharp
TokenType.NoneKeyword => new NoneLiteralSyntax(ExpectKeyword(LanguageConstants.NoneKeyword)),
```

But this change broke the code. It compiles, but when we run it, compilation of a bicepparam file breaks with a message;

```bicepparam
using none

param three = 'param three'
```

```bash
/p/bicep/src/Bicep.Core.Samples/Files/baselines_bicepparam/Extends/shared.bicepparam(1,1) : Error BCP261: A using declaration must be present in this parameters file.
/p/bicep/src/Bicep.Core.Samples/Files/baselines_bicepparam/Extends/shared.bicepparam(1,7) : Error BCP012: Expected the "none" keyword at this location.
```

I'll work on to make it simpler, but for now, I'll leave the code as-is.

Reply:
done, thanks 👍